### ISSUE TITLE
Rake aborted: ModelGen requires Xcode 8.*

### ISSUE THREAD

**Issue by Sajjon** [hebertialmeida/ModelGen#1]
... , but we were not able to find it. If it's already installed update your Spotlight index with 'mdimport /Applications/Xcode*'

I am trying with Xcode 9, is Xcode 9 not supported?

**Comment by hebertialmeida** [comment]
Hey, It is not... I will migrate this to support Xcode 9

**Comment by hebertialmeida** [comment]
It should be working now, let me know if it works for you...

**Comment by Sajjon** [comment]
@hebertialmeida Awesome, thanks! I did a `pod repo update` and got version `0.2.0` now `rake cli:install` starts but it seems getting stuck on `Running script ⚠️ SwiftLint`

Maybe I am doing it wrong? Or maybe I have missunderstood something?

I did not `git clone` the repo. I included it as a pod as you saw above. But the pod does not include a pre-built binary? SwiftGen and Sourcery does. In SwiftGen and Sourcery it is enough to use pods and then call the binary that got included in pods.

How do you feel about including a binary in the pods? Or why do you have pods? :) Is it possible to use pods to download the project but I have to do `rake cli:install` before calling ModelGen, either in the same build phase in my Xcode project, or in two separeate build phases. That increases build time a lot, right?

Anyway, using pods to download the project and then `rake cli:install` gets stuck on Swiftlint scipt:

```
➜  MyProj/Pods/ModelGen git:(master) ✗ rake cli:install
== Building Binary ==
set -o pipefail && ( xcrun xcodebuild -workspace "ModelGen.xcworkspace" -scheme "modelgen" -configuration "Release" -derivedDataPath "<PATH_TO_PROJECT_OMITTED>/Pods/ModelGen/build" MODELGEN_OTHER_LDFLAGS="-sectcreate __TEXT __info_plist <PATH_TO_PROJECT_OMITTED>/Pods/ModelGen/build/Build/Products/Release/modelgen.app/Contents/Info.plist") | bundle exec xcpretty -c
▸ Building Pods/Yams [Release]
▸ Check Dependencies
▸ Processing Info.plist
▸ Compiling api.c
▸ Compiling emitter.c
▸ Compiling parser.c
▸ Compiling reader.c
▸ Compiling scanner.c
▸ Compiling Yams-dummy.m
▸ Compiling writer.c
▸ Compiling Yams_vers.c
▸ Linking Yams
▸ Generating 'Yams.framework.dSYM'
▸ Copying CYaml.h
▸ Copying yaml.h
▸ Copying yaml_private.h
▸ Copying Yams-umbrella.h
▸ Copying Yams.h
▸ Touching Yams.framework
▸ Building Pods/Commander [Release]
▸ Check Dependencies
▸ Processing Info.plist
▸ Compiling Commander-dummy.m
▸ Compiling Commander_vers.c
▸ Linking Commander
▸ Generating 'Commander.framework.dSYM'
▸ Copying Commander-umbrella.h
▸ Touching Commander.framework
▸ Building Pods/PathKit [Release]
▸ Check Dependencies
▸ Processing Info.plist
▸ Compiling PathKit.swift
▸ Compiling PathKit-dummy.m
▸ Compiling PathKit_vers.c
▸ Linking PathKit
▸ Generating 'PathKit.framework.dSYM'
▸ Copying PathKit-umbrella.h
▸ Touching PathKit.framework
▸ Building Pods/Stencil [Release]
▸ Check Dependencies
▸ Processing Info.plist
▸ Compiling Stencil-dummy.m
▸ Compiling Stencil_vers.c
▸ Linking Stencil
▸ Generating 'Stencil.framework.dSYM'
▸ Copying Stencil-umbrella.h
▸ Touching Stencil.framework
▸ Building Pods/StencilSwiftKit [Release]
▸ Check Dependencies
▸ Processing Info.plist
▸ Compiling StencilSwiftKit-dummy.m
▸ Compiling StencilSwiftKit_vers.c
▸ Linking StencilSwiftKit
▸ Generating 'StencilSwiftKit.framework.dSYM'
▸ Copying StencilSwiftKit-umbrella.h
▸ Touching StencilSwiftKit.framework
▸ Building Pods/Pods-modelgen [Release]
▸ Check Dependencies
▸ Processing Info.plist
▸ Compiling Pods-modelgen-dummy.m
▸ Compiling Pods_modelgen_vers.c
▸ Copying Pods-modelgen-umbrella.h
▸ Touching Pods_modelgen.framework
▸ Building ModelGen/modelgen [Release]
▸ Check Dependencies
▸ Processing Info.plist
▸ Running script '[CP] Check Pods Manifest.lock'
▸ Running script '⚠️ SwiftLint'
```

**Comment by Sajjon** [comment]
Maybe I have completely misunderstood something? 😅😂

**Comment by hebertialmeida** [comment]
So, as this is a CLI tool I have not developed an integration to the project yet, it simply installs in your mac and you can run it using `modelgen`.

To install basically you have to clone it somewhere on your machine. In the future, I plan to create finish the pod integration that will run whenever you build your project, but for not just install on your machine 😄

```bash
$ git clone https://github.com/hebertialmeida/ModelGen.git
$ cd ModelGen
$ rake cli:install
```

Let me know if you could generate some models, I have a project here with ~60 immutable models and it saves a lot of time and keeps it consistent.

**Comment by Sajjon** [comment]
Okay thanks for clairification. Maybe I will fork the repo, build a binary and include that as Sourcery does in my fork :)

**Comment by hebertialmeida** [comment]
Nice, could you do a Pull Request for that implementation? So others can use it 👍

**Comment by Sajjon** [comment]
I am on it!

**Comment by Sajjon** [comment]
I did not succeed unfortunately, Maybe you can have a look at it @hebertialmeida ?

I do not know how SwiftLint, SwiftGen and Sourcery does it, do they create and upload the binary as part of the Rakefile? Are they using `lipo` tool to create one fat library? I had som problems with binary not finding Commander framework. When the binary is in a folder `bin`, with another folder `lib` next to that folder, where `lib` contained all frameworks.

It looks like you are doing some packaging of all the frameworks in [these lines in the RakeFile](https://github.com/hebertialmeida/ModelGen/blob/master/Rakefile#L57-L68) but still I got the error that `Commander.framework` was not found.

Sorry that I was unable to fix it, can you have a look? <3

**Comment by hebertialmeida** [comment]
You can open a Pull Request and I can try to help to implement that. The better way is to add it to homebrew, but I can't because I am the owner... This is a rule on their side 😞

**Comment by Sajjon** [comment]
@hebertialmeida But cant we expand the rakefile of ModelGen to make it possible to upload a binary to Github? So that we have bundled releases and so that a project `MyProject` that wants to use `ModelGen` can do it via cocoapods and then call the bundled binary using `$PODS_ROOT/ModelGen/bin/modelgen` similar to how [Sourcery does it](https://github.com/krzysztofzablocki/Sourcery#installation)?

**Comment by hebertialmeida** [comment]
That's an interesting way to do, would you be able to implement that?

**Comment by Sajjon** [comment]
I could give it a try again later on, maybe will have time during March

**Comment by hebertialmeida** [comment]
This can be closed as I'm now supporting swift 4.2 + Xcode 10.1
